avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,311 @@
+                              performed during the sysinfo collection.
+        :type sysinfo_files: dict
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        """
+        self.config = config
+        self.queue = queue
+        self._installed_pkgs = None
+        packages_namespace = 'sysinfo.collect.installed_packages'
+        self.log_packages = self.config.get(packages_namespace, False)
+
+        self.sysinfo_files = sysinfo_files
+        self._set_collectibles()
+
+    def _set_collectibles(self):
+        self.collectibles = set()
+        for cmd in self.sysinfo_files.get("commands", []):
+            self.collectibles.add(_Command(cmd))
+
+        for filename in self.sysinfo_files.get("files", []):
+            self.collectibles.add(_Logfile(filename))
+
+    def _get_installed_packages(self):
+        sm = software_manager.SoftwareManager()
+        installed_pkgs = sm.list_all()
+        self._installed_pkgs = installed_pkgs
+        return installed_pkgs
+
+    def collect(self):
+        """Log all collectibles at the start of the event."""
+        for log_hook in self.collectibles:
+            log_hook.run(self.queue, self.sysinfo_dir)
+
+        if self.log_packages:
+            self._log_packages(self.sysinfo_dir)
+        self.queue.put(messages.FinishedMessage.get('pass'))
+
+    def _log_packages(self, path):
+        installed_path = os.path.join(path, "installed_packages")
+        installed_packages = "\n".join(self._get_installed_packages()) + "\n"
+        self.queue.put(messages.FileMessage.get(installed_packages,
+                                                installed_path))
+
+
+class PostSysInfo(PreSysInfo):
+    """
+    Log different system properties after end event.
+
+    An event may be a job, a test, or any other event with a
+    beginning and end.
+    """
+
+    sysinfo_dir = os.path.join('sysinfo', 'post')
+
+    def __init__(self, config, sysinfo_files, queue, test_fail=False):
+        """
+        :param test_fail: flag for fail tests. Default False
+        :type test_fail: bool
+        """
+        self.test_fail = test_fail
+        super().__init__(config, sysinfo_files, queue)
+
+    def _set_collectibles(self):
+        super()._set_collectibles()
+        if self.test_fail:
+            for fail_cmd in self.sysinfo_files.get("fail_commands", []):
+                self.collectibles.add(_Command(fail_cmd))
+
+            for fail_filename in self.sysinfo_files.get("fail_files", []):
+                self.collectibles.add(_Logfile(fail_filename))
+
+
+class SysinfoRunner(nrunner.BaseRunner):
+    """
+    Runner for gathering sysinfo
+
+    Runnable attributes usage:
+
+     * uri: Specification of sysinfo. Possible values pre/post
+
+     * kwargs: sysinfo_files dictionary with commands/tasks which should be
+             performed during the sysinfo collection.
+    """
+
+    def run(self):
```

Comment:
It would be good to catch empty arguments and display an error message. Example:

```
$ avocado-runner-sysinfo runnable-run
{'status': 'started', 'time': 60072.442789556}
{'type': 'stderr', 'log': b'Traceback (most recent call last):\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/sysinfo.py", line 270, in run\n    sysinfo = PostSysInfo(self.runnable.config,\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/sysinfo.py", line 235, in __init__\n    super().__init__(config, sysinfo_files, queue)\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/sysinfo.py", line 187, in __init__\n    self._set_collectibles()\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/sysinfo.py", line 238, in _set_collectibles\n    super()._set_collectibles()\n  File "/home/wrampazz/src/avocado/avocado.dev/avocado/core/runners/sysinfo.py", line 191, in _set_collectibles\n    for cmd in self.sysinfo_files.get("commands", []):\nAttributeError: \'NoneType\' object has no attribute \'get\'\n', 'encoding': 'utf-8', 'status': 'running', 'time': 60072.445269423}
{'result': 'error', 'status': 'finished', 'time': 60072.445292247}
```